client/core/core.go

```diff
@@ -1573,15 +1573,14 @@ func (c *Core) WalletSettings(assetID uint32) (map[string]string, error) {
 }
 
 // ReconfigureWallet updates the wallet configuration settings, it also updates
-// the password if newWalletPW is non-nil.
+// the password if newWalletPW is non-nil. Do not make concurrent calls to
+// ReconfigureWallet for the same asset.
 func (c *Core) ReconfigureWallet(appPW, newWalletPW []byte, assetID uint32, cfg map[string]string) error {
 	crypter, err := c.encryptionKey(appPW)
 	if err != nil {
 		return newError(authErr, "ReconfigureWallet password error: %v", err)
 	}
-	c.walletMtx.Lock()
-	defer c.walletMtx.Unlock()
-	oldWallet, found := c.wallets[assetID]
+	oldWallet, found := c.wallet(assetID)
```

Comment:
I did not see a solid reason for the function to guard the wallets map from access to update, other than to ensure proper results if ReconfigureWallet were called for the same asset several times concurrently.  Resolved with docs.

Reply:
> other than to ensure proper results if ReconfigureWallet were called for the same asset several times concurrently

Yes, that was the reasononing.

Reply:
How about if an attempt is made to access this wallet while this reconfiguration is going on? I do understand not preventing access to all wallets in order to reconfigure one but I do think the wallet being reconfigured should be locked for access.

Reply:
Problem is there are already numerous places that just grab the xcWallet with the `wallet()` method and move on to using it, so if one of those started and then ReconfigureWallet happened, even keeping this locked for the duration won't prevent the other goroutine from getting their wallet disconnected unexpectedly.

Reply:
oh, noted

Reply:
I've been starting to suspect recently that we need two types of reconfigure: (1) one that affects actually wallet connection like RPC params, and (2) other random options that can just be applied to the existing wallet backend with target mutator methods.

Reply:
Only the asset developer really knows which settings require reconnect, so I'd recommend we add a `Reconfigure(cfg map[string]string)` method to `asset.Wallet` instead.

Reply:
Right, I didn't mean two different Core methods necessarily.

Although the ReconfigureWallet interface is unfortunately cumbersome for many settings changes so I'd not rule out convenience methods that avoid the need to get the current settings map only to feed it back with maybe one tweak to it.  A different idea entirely though.